subsys/bluetooth/host/l2cap.c

```diff
@@ -2343,6 +2271,8 @@ int bt_l2cap_chan_recv_complete(struct bt_l2cap_chan *chan, struct net_buf *buf)
 	__ASSERT_NO_MSG(chan);
 	__ASSERT_NO_MSG(buf);
 
+	net_buf_unref(buf);
```

Comment:
Is this correct? On line 2287 you're passing the buffer to `l2cap_chan_send_credits()` so you must hold a valid reference to the buffer until then.

Reply:
yes. `buf` is overwritten in that fn. It's passed to `l2cap_create_le_sig_pdu` but then it's also overwritten. 
I guess it was part of some sort of optimization or buffer re-use before.
I had a commit that removed passing this useless `buf` around but de-scoped it from the PR.